fe/lib/components/SalaryDetails/v2/SalaryDetails.tsx

```diff
@@ -0,0 +1,230 @@
+import {
+  Box,
+  Column,
+  Columns,
+  Dropdown,
+  Heading,
+  RadioGroup,
+  RadioItem,
+  Stack,
+  Text,
+  TextField,
+} from 'braid-design-system';
+import React, { useId, useState } from 'react';
+
+import {
+  type PayRangeChange,
+  type PayType,
+  type PayTypeChange,
+  SALARY_CURRENCIES,
+  type SalaryCurrency,
+  type SalaryCurrencyChange,
+  type SalaryDescriptionChange,
+  type SalaryError,
+  payTypes,
+} from './types';
+import {
+  validateDescription,
+  validateMaxAmount,
+  validateMinAmount,
+} from './validateSalary';
+
+import * as styles from './styles.css';
+
+export const MAX_CHAR_LIMIT = 50;
+
+export interface SalaryDetailsProps {
+  errors?: SalaryError;
+  initialCurrency?: SalaryCurrency;
+  initialDescription?: PayType;
+  initialMaximumAmount?: PayType;
+  initialMinimumAmount?: PayType;
+  initialPayType?: PayType;
+  onBlur: (
+    item:
+      | PayRangeChange
+      | PayTypeChange
+      | SalaryCurrencyChange
+      | SalaryDescriptionChange,
+  ) => void;
+}
+
+export const SalaryDetails = (props: SalaryDetailsProps) => {
+  const id = useId();
+
+  const {
+    errors,
+    initialCurrency,
```

Comment:
Should we add a `useEffect` in for these initial values? I think this could be useful, e.g., if someone changes the location of their job and we want to set a new initial currency

Reply:
I've given this a shot in [49bd098](https://github.com/seek-oss/wingman/pull/935/commits/49bd098a13ad66cc8c1ee231d309afbf03a752d6); the way our internal codebase currently integrates this component may make things challenging but let's see how it works out 🙂